### PR TITLE
added responsive nav bar

### LINKED ISSUE #54: Create mobile navigation based on desktop one
The mockups have been created as a desktop, but eventually we'll need a mobile version of the site, and it's good to do it parallel. Does Agile Ventures have some icon library you use already, maybe @FedericoEsparza you know? I would suggest the [Noun project](https://thenounproject.com/) as they have really nice icons.

### PR DESCRIPTION
**Pull request by jmpainter**
- Added Layout container with nav bar state
- moved Header and Footer into components/Layout
- created NavBar component

tests have not been updated yet

fixes #54

### CODE REVIEW

**Comment by yakryder** [comment]
While it seems we don't have anything in the PR review process that prevents it, I'm exercising the right not to review this until the build is green. That shouldn't prevent anyone else who wants to do so though

**Comment by FedericoEsparza** [comment]
Yep. Reviewing a PR can be potentially time-consuming, so I don't like to even look at PRs that are failing CI or labeled as [WIP].

UNLESS it's a languishing PR, not going anywhere, or creating a pile-up or bottleneck, in which case I might jump **in--and I would encourage others to feel free--to try to fix the PR, preferably with a PR to the original PR's branch.** Just be respectful of the contributor's effort. 

While I do want to adhere to AV's mission to help developers learn to program and gain experience on real projects, and I think we go out of our way to help foster that environment in Championer projects, AV's partnership with Championer doesn't preclude us from helping to move things along, when called for.

@jmpainter I assume you got this under control, but let me know if you get stuck.

**Comment by jmpainter** [comment]
I updated the tests for a passing build. I'll add additional tests once the basic approach of this PR is confirmed.

File: src/containers/Layout/Layout.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  handleClick() {
+    this.setState({
+      mobileMenu: !this.state.mobileMenu
+    })
+  }
+
+  render() {
+    return (
+      <div className="main-container">
```
Comment (FedericoEsparza) [inline src/containers/Layout/Layout.js]: I think you'll want remove `className="main-container"` in order for the Header to reach the full width of the page.

File: src/components/App.js [hunk 2]
```diff
@@ -14,15 +15,12 @@ import Artist from '../containers/Artist/Artist'
...
-import Footer from './Footer/Footer'
 
 const App = () => {
   return (
     <BrowserRouter>
-      <div>
-        <Header />
-        <main className="main-container grid">
+      <Layout>
+        <main className="grid">
```
Comment (FedericoEsparza) [inline src/components/App.js]: I think you'll want to re-add the `main-container` class to `<main className="grid">` here.

File: src/components/App.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import React from 'react'
 import { BrowserRouter, Route } from 'react-router-dom'
+import Layout from '../containers/Layout/Layout'
```
Comment (FedericoEsparza) [inline src/components/App.js]: You can move the Layout down with the other components.

File: src/containers/Layout/Layout.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import PropTypes from 'prop-types'
+
+export default class Layout extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      mobileMenu: false
+    }
+    this.handleClick = this.handleClick.bind(this)
+  }
```
Comment (FedericoEsparza) [inline src/containers/Layout/Layout.js:9]: Let's get rid of the constructor, and use the newer state initializer syntax, with or without the semicolon, as so:
Screen Shot 2019-04-26 at 11 21 15 PM

File: src/components/Layout/Header/header.scss [hunk 5]
```diff
@@ -0,0 +1,6 @@
+.header-container {
+  width: 100%;
+  background-color: $primary-color-darkblue;
```
Comment (FedericoEsparza) [inline src/components/Layout/Header/header.scss]: You'll want to keep this as `$primary-color-darkblue-lighter;`

**Review by FedericoEsparza — CHANGES_REQUESTED** [review]
@jmpainter You're doing an amazing job, heading in the right direction, Josh. 

I like how the mobile dropdown mobileMenu looks.

I think we'll want to figure out a solution for not having the NavLinks stack, as the window narrows on a desktop, though.

Let me know if you have trouble trying to fix this.

See the line comments about removing the constructor for the `Layout` container. This will cause the linter to complain, but it should be ok, once ticket #188 is merged, and you pull from upstream.

Please, also move the `NavBar` component folder into the `src/components/Layout/` folder.

**Keep at it! You're doing great work with this PR.**

File: src/containers/Layout/Layout.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+  handleClick() {
+    this.setState({
+      mobileMenu: !this.state.mobileMenu
+    })
+  }
```
Comment (FedericoEsparza) [inline src/containers/Layout/Layout.js:15]: Please refactor to initialize with an arrow function, as so:
Screen Shot 2019-04-28 at 11 57 47 PM
Doing so will keep us from needing to bind `this` for the function, and we can use the state initializer syntax, without a constructor, which is more clean.

**Comment by jmpainter** [comment]
Thanks for your feedback on this PR. I made the requested changes and added more tests. I'm keeping the breakpoints for the NavBar consistent with those in _layout.scss. The build is failing CI due to eslint.

**Review by FedericoEsparza — APPROVED** [review]

**Comment by FedericoEsparza** [comment]
@jmpainter Absolutely excellent work, Josh.

I know it was a lot of work, but it's looking great. And you fixed all the requested changes.

I went ahead and re-added the navlinks to the footer, and updated the copyright message:
Screen Shot 2019-05-12 at 3 15 41 AM

We'll have to figure out a way to make them more mobile friendly, but I didn't want them to get left behind.

I think we'll end up going with a more vertically stacked footer, with more links, in the end.

Excellent job wrangling the mobile header to play nice with the desktop header and with the whole reconstructed Layout with props for the header and footer:
Screen Shot 2019-05-12 at 3 22 48 AM

**Pull request merged**